sdk/batch/batch-rest/MigrationGuide.md

```diff
@@ -0,0 +1,265 @@
+
+```typescript
+import { DefaultAzureCredential } from "@azure/identity";
+import createClient from "@azure-rest/batch";
+
+const credential = new DefaultAzureCredential();
+const client = createClient("<account-endpoint>", credential);
+```
+
+## Operation response differences
+
+Previously in `@azure/batch`, the client operation return a `Promise` that resolves to the result of the response body JSON. The following example demonstrates how to get a job with the `BatchServiceClient` instance.
+
+```typescript
+const job = await client.job.get("<job-id>");
+console.log(`Job id: ${job.id}, state: ${job.state}`);
+```
+
+Now in `@azure-rest/batch`, the client operation return a `Promise` that resolves to the response object, which contains the response body and the status code. In order to get the response body JSON, you need to first check if the response is unexpected with the `isUnexpected` helper method, then access the response body. The following example demonstrates how to get a job in `@azure-rest/batch`.
+
+```typescript
+import { isUnexpected } from '@azure-rest/batch';
+const response = await client.path("/jobs/{jobId}", "<job-id>").get();
+if (isUnexpected(response)) {
+  throw new Error(`Failed to get job: ${response.body.message}`);
+}
+console.log(`Response status code: ${response.status}`);
+
+const job = response.body;
+console.log(`Job id: ${job.id}, state: ${job.state}`);
+```
+
+## Error handling
+
+Previously in `@azure/batch`, the client operation succeed only when service return expected HTTP status code, for example `201` for create resources operations or `200` for general HTTP GET requests. Unexpected HTTP status code will throw a `RestError` from `@azure/ms-rest-js` package. The following example demonstrate how to handle different errors might occur in the get pool request.
+
+```typescript
+import { RestError } from "@azure/ms-rest-js";
+
+try {
+  const pool = await client.pool.get("<pool-id>");
+  console.log("Get pool success: ", pool)
+} catch (error) {
+  if (error instanceof RestError) {
+    // Returned HTTP status is not 200
+    console.log(`Service return unexpected status code ${error.statusCode}: ${error.body}`)
+  } else {
+    // Other errors like connection errors or other exceptions
+    console.log("Failed to get pool with error: ", error)
+  }
+}
+```
+
+Now, for `@azure-rest/batch`, the client operation won't throw errors even when the returned HTTP status code is unexpected, instead it exports a helper method `isUnexpected` to help you check if the response is unexpected. The following example demostarte how to handle different errors might occur in the get pool request.
```

Comment:
Suggestion:

In `@azure-rest/batch`, the client operation won't throw errors even when the returned HTTP status code is unexpected. Instead it exports a helper method `isUnexpected` to help you check if the response is unexpected. The following example demonstrates how to handle different errors which might occur in the get pool request.